### PR TITLE
*: use kvproto::backup::StorageBackend in create_storage

### PR DESCRIPTION
**Pull request by kennytm**
###  What have you changed?

Please explain in detail what the changes are in this PR and why they are needed:

- In pingcap/kvproto#507, we have removed the string-based "path" and "url" field from `backup::BackupRequest` and `import_sstpb::DownloadRequest` respectively, and replaced them with a structural type `backup::StorageBackend`, so that storage with multiple options (e.g. S3) can be passed without unnecessary deserialization.
- This PR updates `kvproto` to include that change. 
- The `external_storage::create_storage` function is updated to take a `&backup::StorageBackend` instead of `&str`
- The `backup` and `sst_importer` components and test cases are updated accordingly.

###  What is the type of the changes?

Pick one of the following and delete the others:

- Engineering (engineering change which doesn't change any feature or fix any issue)

###  How is the PR tested?
Please select the tests that you ran to verify your changes:
- Unit test

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?

No

###  Does this PR affect `tidb-ansible`?

No

###  Refer to a related PR or issue link (optional)

pingcap/kvproto#507
pingcap/br#88

### CODE REVIEW

**Comment by kennytm** [comment]
PTAL @overvenus @5kbpers @tennix @DanielZhangQD

File: components/external_storage/src/lib.rs [hunk 1]
```diff
@@ -14,38 +14,94 @@ use std::io::{self, Read};
...
+                warn!("ignoring invalid S3 bucket name"; "bucket" => &s3.bucket, "error" => %e);
+            }
+            u.set_path(s3.get_prefix());
+        }
+        None => {}
+    }
+    u
+}
+
+/// Creates a local `StorageBackend` to the given path.
```
Comment (DanielZhangQD) [inline components/external_storage/src/lib.rs]: S3 backend is not covered in this PR?

Comment (kennytm) [reply components/external_storage/src/lib.rs]: Right, that would be another PR.

**Pull request merged**